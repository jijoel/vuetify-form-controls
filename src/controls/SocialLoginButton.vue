<template>
<v-tooltip top>
  <v-btn icon slot="activator" @click="click">
    <v-icon :color="_color">
      {{ _icon }}
    </v-icon>
  </v-btn>
  <span>
    Login with {{ name }}
  </span>
</v-tooltip>
</template>

<script>
import axios from 'axios'

export default {

  name: 'SocialLogin',

  props: {
    color: {
      type: String,
    },
    icon: {
      type: String,
    },
    name: {
      type: String,
      required: true,
    },
    path: {
      type: String,
      required: false,
      default: '/api/oauth/'
    }
  },

  computed: {
    _color() {
      return this.color
      || {
        facebook:   '#3B5998',
        instagram:  '#262626',
        twitter:    '#1DA1F2',
        pinterest:  '#C92228',
        snapchat:   '#FFFC00',
        google:     '#DB4437',
        github:     '#666'
      }[this.name]
      || '#666'
    },
    _icon() {
      return this.icon
      || {
        facebook:   'fab fa-facebook-square',
        instagram:  'fab fa-instagram',
        twitter:    'fab fa-twitter',
        pinterest:  'fab fa-pinterest',
        snapchat:   'fab fa-snapchat',
        google:     'fab fa-google-plus',
        github:     'fab fa-github'
      }[this.name]
      || 'fas fa-user-circle'
    },
    _path() {
      return this.path + this.name
    },
  },

  methods: {
    async click() {
      const { data } = await axios.get(this._path)
      window.location = data.url
    }
  },

}
</script>
