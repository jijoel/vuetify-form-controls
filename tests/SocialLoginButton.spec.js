import './setup'
import { shallowMount } from '@vue/test-utils'
import SocialLoginButton from '../src/controls/SocialLoginButton'

describe('SocialLoginButton', () => {

  const mocks = { '$t': (val) => val }

  test('it renders a standard button', () => {
    const wrapper = shallowMount(SocialLoginButton, {
        propsData: { name: 'foo' },
        mocks,
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('it renders a github button', () => {
    const wrapper = shallowMount(SocialLoginButton, {
        propsData: { name: 'github' },
        mocks,
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('it renders a facebook button', () => {
    const wrapper = shallowMount(SocialLoginButton, {
        propsData: { name: 'facebook' },
        mocks,
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  // TODO: Mock the axios plugin @/plugins/axios
  test.skip('it sets the window location to the social login page', ()=>{
  })

})
