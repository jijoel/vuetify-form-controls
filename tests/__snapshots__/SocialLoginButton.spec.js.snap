// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLoginButton it renders a facebook button 1`] = `
<v-tooltip-stub
  closedelay="200"
  debounce="0"
  fixed="true"
  opendelay="200"
  tag="span"
  top=""
>
  <v-btn-stub
    activeclass="v-btn--active"
    icon="true"
    tag="button"
    type="button"
  >
    <v-icon-stub
      color="#3B5998"
    >
      
      fab fa-facebook-square
    
    </v-icon-stub>
  </v-btn-stub>
   
  <span>
    
    login_with facebook
  
  </span>
</v-tooltip-stub>
`;

exports[`SocialLoginButton it renders a github button 1`] = `
<v-tooltip-stub
  closedelay="200"
  debounce="0"
  fixed="true"
  opendelay="200"
  tag="span"
  top=""
>
  <v-btn-stub
    activeclass="v-btn--active"
    icon="true"
    tag="button"
    type="button"
  >
    <v-icon-stub
      color="#666"
    >
      
      fab fa-github
    
    </v-icon-stub>
  </v-btn-stub>
   
  <span>
    
    login_with github
  
  </span>
</v-tooltip-stub>
`;

exports[`SocialLoginButton it renders a standard button 1`] = `
<v-tooltip-stub
  closedelay="200"
  debounce="0"
  fixed="true"
  opendelay="200"
  tag="span"
  top=""
>
  <v-btn-stub
    activeclass="v-btn--active"
    icon="true"
    tag="button"
    type="button"
  >
    <v-icon-stub
      color="#666"
    >
      
      fas fa-user-circle
    
    </v-icon-stub>
  </v-btn-stub>
   
  <span>
    
    login_with foo
  
  </span>
</v-tooltip-stub>
`;
